airbyte-integrations/connectors/source-stripe/source_stripe/source.py

```diff
@@ -97,6 +102,56 @@ def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) ->
             return False, str(e)
         return True, None
 
+    @staticmethod
+    def is_test_account(config: Mapping[str, Any]) -> bool:
+        """Check if configuration uses Stripe test account (https://stripe.com/docs/keys#obtain-api-keys)
+
+        :param config:
+        :return: True if configured to use a test account, False - otherwise
+        """
+
+        return str(config["client_secret"]).startswith(STRIPE_TEST_ACCOUNT_PREFIX)
+
+    def get_api_call_budget(self, config: Mapping[str, Any]) -> APIBudget:
+        """Get API call budget which connector is allowed to use.
+
+        :param config:
+        :return:
+        """
+
+        max_call_rate = 25 if self.is_test_account(config) else 100
+        if config.get("call_rate_limit"):
+            call_limit = config["call_rate_limit"]
+            if call_limit > max_call_rate:
```

Comment:
Should we have `call_limit = max_call_rate` in that case?

Reply:
Adding more information: If `config["call_rate_limit"] = 200`, it seems like the `FixedWindowCallRatePolicy` without matchers will have a call_limit of 200 which is bigger than the max allowed

Reply:
right, I absolutely missed this part 
actually it is in in another change set merged with this PR https://github.com/airbytehq/airbyte/pull/31696
I'm waiting for it to be merged to update this.

Reply:
The changes that @yevhenii-ldv did include moving from HttpStream to HttpRequester which needs to receive call_budget with proper rate limited Session. 
But because his PR is not going anywhere soon, I think this PR is not blocked anymore.